File: fly-http/src/main/java/com/xia/fly/http/utils/ParamsUtils.java
package com.xia.fly.http.utils;

import java.util.Map;

/**
 * @author xia
 * @date 2018/10/8.
 * <p>
 * 处理空参数
 */
@SuppressWarnings("UnusedReturnValue")
public final class ParamsUtils {

    private ParamsUtils() {
        throw new IllegalStateException("u can't instantiate me!");
    }

    public static void escapeParams(Map<String, String> map) {
        if (map != null && !map.isEmpty()) {
            for (Map.Entry<String, String> entry : map.entrySet()) {
                String value = entry.getValue();
                value = value == null ? "" : value;
                map.put(entry.getKey(), value);
            }
        }
    }
}

File: fly-http/src/main/java/com/xia/fly/http/utils/ParamsUtils.kt
package com.xia.fly.http.utils

/**
 * @author xia
 * @date 2018/10/8.
 *
 *
 * 用于处理空参数
 */
object ParamsUtils {

    @JvmStatic
    fun escapeParams(map: MutableMap<String, String>?) {
        if (map?.isNotEmpty() == true) {
            for (entry in map.entries) {
                val value: String? = entry.value
                map[entry.key] = value ?: ""
            }
        }
    }
}
